src/modules/producer.js

```diff
@@ -0,0 +1,150 @@
+      .then(() => {
+        delete rpcQueue[corrIdA];
+      })
+      .catch(this.conn.config.transport.error);
+  };
+}
+
+function publishOrSendToQueue(queue, msg, options) {
+  if (!options.routingKey) {
+    return this.channel.sendToQueue(queue, msg, options);
+  } else {
+    return this.channel.publish(queue, options.routingKey, msg, options);
+  }
+}
+
+/**
+ * Send message with or without rpc protocol, and check if RPC queues are created
+ * @param  {string} queue   the queue to send `msg` on
+ * @param  {any} msg     string, object, number.. anything bufferable/serializable
+ * @param  {object} options contain rpc property (if true, enable rpc for this message)
+ * @return {Promise}         Resolves when message is correctly sent, or when response is received when rpc is enabled
+ */
+function checkRpc (queue, msg, options) {
+  //messages are persistent
+  options.persistent = true;
+
+  if (options.rpc) {
+    if (!amqpRPCQueues[queue]) {
+      amqpRPCQueues[queue] = {};
+    }
+
+    return createRpcQueue.call(this, queue)
+    .then(() => {
+      //generates a correlationId (random uuid) so we know which callback to execute on received response
+      var corrId = uuid.v4();
+      options.correlationId = corrId;
+      //reply to us if you receive this message!
+      options.replyTo = amqpRPCQueues[queue].queue;
+
+      publishOrSendToQueue.call(this, queue, msg, options);
+
+      //defered promise that will resolve when response is received
+      amqpRPCQueues[queue][corrId] = Promise.defer();
+      return amqpRPCQueues[queue][corrId].promise;
+    });
+  }
+
+  return publishOrSendToQueue.call(this, queue, msg, options);
+}
+
+/**
+ * Ensure channel exists and send message using `checkRpc`
+ * @param  {string} queue   The destination queue on which we want to send a message
+ * @param  {any} msg     Anything serializable/bufferable
+ * @param  {object} options message options (persistent, durable, rpc, etc.)
+ * @return {Promise}         checkRpc response
+ */
+function produce(queue, msg, options) {
+  return this.conn.get()
+  .then((_channel) => {
+    this.channel = _channel;
+
+    //undefined can't be serialized/buffered :p
+    if (!msg) msg = null;
+
+    //default options are persistent and durable because we do not want to miss any outgoing message
+    //unless user specify it
+    options = options || { persistent: true, durable: true };
+
+    this.conn.config.transport.info('amqp:producer', '[' + queue + '] > ', msg);
+
+    return checkRpc.call(this, queue, parsers.out(msg, options), options);
+  })
+  .catch((err) => {
+    //add timeout between retries because we don't want to overflow the CPU
+    this.conn.config.transport.error('amqp:producer', err);
+    return utils.timeoutPromise(this.conn.config.timeout)
+    .then(() => {
+      return produce.call(this, queue, msg, options);
+    });
+  });
+}
+
+module.exports = function(_conn) {
+  return {
+    conn: _conn,
+    produce: produce
```

Comment:
I think property shorthands are working so if the have the same name just  ` {conn: _conn, produce}` should work.